src/python/engagement_edge/tests.Dockerfile

```diff
@@ -0,0 +1,16 @@
+# Motivation for a different `-test` image:
```

Comment:
We currently install `hypothesis`, `pytest`, `pytest-xdist`, and some other things in the `grapl-python-build` image (and therefore in the virtualenv, and consequently `/site-packages` in the Lambda) of every production service. Would it make sense to make the split at that level too s.t. these deps don't end up in the prod artifacts?

If you unzip one of our deployment artifacts, for example, there are all these test libraries in it. It would be better for each deployment image (and therefore each Lambda artifact) to have only the libraries needed to run the service.

Reply:
This is done universally in #444